utilities.js

```diff
@@ -58,11 +58,21 @@ exports.equals = function(x, y) {
         if (y.hasOwnProperty(p) && !x.hasOwnProperty(p)) return false;
     }
     return true;
-  }
+}
+
+exports.calculateMatchLength = function(duration) {
+    const time = Math.round(duration)
+    const hours = Math.floor(time / 3600)
+    const remainingSeconds = time - hours * 3600
+
+    const minutes = Math.floor(remainingSeconds / 60)
+    const seconds = String(remainingSeconds - minutes * 60).padStart(2, "0")
 
-  exports.secondsToMinuteSeconds = function(duration) {
-      const minutes = Math.round(duration / 60)
-      const seconds = Math.round(duration % 60)
+    return {
+        hours,
+        minutes: padTime(minutes),
+        seconds: padTime(seconds)
```

Comment:
think seconds gets padded twice

Reply:
You're right i forgot to remove it from `L69`, will update